File: typeOfTriangle.py
from typing import List

class Solution:
    def triangleType(self, nums: List[int]) -> str:
        a, b, c = sorted(nums)
        # Check for triangle inequality
        if a + b <= c:
            return "none"
        # Check for equilateral triangle
        if a == b == c:
            return "equilateral"
        # Check for isosceles triangle
        if a == b or b == c or a == c:
            return "isosceles"
        # If all sides are different
        return "scalene"

File: typeOfTriangle.go
import "sort"

func triangleType(nums []int) string {
    sort.Ints(nums)
    a, b, c := nums[0], nums[1], nums[2]

    // Check triangle inequality
    if a + b <= c {
        return "none"
    }

    // Check for equilateral triangle
    if a == b && b == c {
        return "equilateral"
    }

    // Check for isosceles triangle
    if a == b || b == c || a == c {
        return "isosceles"
    }

    // All sides different
    return "scalene"
}
